npm/private/npm_package_store.bzl

```diff
@@ -189,14 +188,16 @@ def _npm_package_store_impl(ctx):
 
     # the path to the package store location for this package
     # "node_modules/{package_store_root}/{package_store_name}/node_modules/{package}"
-    package_store_directory_path = paths.join("node_modules", utils.package_store_root, package_store_name, "node_modules", package)
+    package_store_directory_path = "node_modules/{}/{}/node_modules/{}".format(utils.package_store_root, package_store_name, package)
 
     if ctx.attr.src and NpmPackageInfo in ctx.attr.src:
         # output the package as a TreeArtifact to its package store location
         if ctx.label.workspace_name:
-            expected_short_path = paths.join("..", ctx.label.workspace_name, ctx.label.package, package_store_directory_path)
+            expected_short_path = "../{}/{}/{}".format(ctx.label.workspace_name, ctx.label.package, package_store_directory_path)
+        elif ctx.label.package:
+            expected_short_path = "{}/{}".format(ctx.label.package, package_store_directory_path)
         else:
-            expected_short_path = paths.join(ctx.label.package, package_store_directory_path)
```

Comment:
in my repo `ctx.label.package` was always an empty string. I suspect its because we only have a single `:node_modules` target at the repo root. I added the case (`elif ctx.label.package`) above to preserve existing behavior, in case it's needed for non-root :node_modules.